bin/reorder-keys

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+if [ ! -d exercises ] ; then
+    echo "\"exercises\" directory not found, make sure you're running from the repository root" >&2
+    exit 1
+fi
+
+for canonical_data_file in exercises/*/canonical-data.json ; do
+    jq 'walk(if (type == "object") and has("uuid") then . as $obj | reduce ["uuid", "reimplements", "description", "comments", "scenarios", "property", "input", "expected"][] as $key ({}; . + (if $obj | has($key) then { ($key): $obj[$key] } else {} end)) else . end)' "$canonical_data_file" > tmp && mv tmp "$canonical_data_file"
```

Comment:
Nit: we could consider:
- Splitting this long line.
- Renaming `tmp` to something a bit more unique. In the unlikely case that some user has a file named `tmp` in their prob-specs repo and runs this script, this script will overwrite it.

Reply:
I've renamed `tmp` to something more unique.

As for the splitting of the long line, do you have any good suggestion on how to format it? I've tried several things, but they were all still fairly unwieldy.

Reply:
> As for the splitting of the long line, do you have any good suggestion on how to format it?

Maybe something like the below:

```shell
#!/usr/bin/env bash

set -e

if [ ! -d exercises ] ; then
    echo "Error: \"exercises\" directory not found, make sure you're running from the repository root" >&2
    exit 1
fi

tmp_file='.reordered-canonical-data.json'

reorder() {
    jq 'walk(if (type == "object") and has("uuid") then . as $obj
        | reduce ["uuid", "reimplements", "description", "comments", "scenarios", "property", "input", "expected"][] as $key(
            {}; . + (if $obj | has($key) then { ($key): $obj[$key] } else {} end)) else . end
        )' "$1" > "$tmp_file"
    mv "$tmp_file" "$1"
}

for canonical_data_file in exercises/*/canonical-data.json ; do
    reorder "$canonical_data_file"
done
```

But it's fine. PR it if you want.

The default shell for GitHub Actions on non-Windows is bash with the `-e` option (actually `bash --noprofile --norc -eo pipefail`, see [docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell)), so I have a slight preference for adding that at the top of the script to get consistent behavior locally. Then we can put the `mv` on a separate line without the `&&`.

(No strong opinion about the extra function).

Reply:
Sorry for the potentially early merge. I saw the `tmp` issue being addressed but missed the refactoring question.

Reply:
No problem.